packages/bsky/src/api/app/bsky/feed/getFeed.ts

```diff
@@ -220,6 +220,16 @@ const skeletonFromFeedGen = async (
         headers,
       },
     )
+
+    const { cursor: resultCursor } = result.data
+    if (params.cursor && resultCursor === params.cursor) {
+      // Prevents loops in custom feeds.
+      throw new UpstreamFailureError(
+        'feed provided an invalid response',
+        'UpstreamFailureError',
+      )
```

Comment:
How would you feel about failing open instead?  If the cursor is the same, we treat it as empty rather than failing.